src/main/java/hanshyn/onlinebookstore/controller/CategoryController.java

```diff
@@ -0,0 +1,77 @@
+package hanshyn.onlinebookstore.controller;
+
+import hanshyn.onlinebookstore.dto.book.BookDto;
+import hanshyn.onlinebookstore.dto.category.CategoryResponseDto;
+import hanshyn.onlinebookstore.dto.category.CreateCategoryRequestDto;
+import hanshyn.onlinebookstore.service.book.BookService;
+import hanshyn.onlinebookstore.service.category.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Categories", description = "Endpoints for category")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @Operation(summary = "Get all categories", description = "Get all categories by store")
+    @PreAuthorize("hasAnyAuthority('ROLE_USER', 'ROLE_ADMIN')")
```

Comment:
You have .anyRequest().authenticated() in your config
This means that unauthorized user cannot access this endpoint. So if user with any tole may access then there is no need to mention them all.
We put annotation only if we want not all of our roles to have access
```suggestion
```
Please recheck all occurences